### PR TITLE
Use device attributes instead of metadata

### PR DESCRIPTION
**Pull request by Pavinati**
Parse device attributes field in place of removed metadata field

Signed-off-by: Mattia <mattia.pavinati@ispirata.com>

### CODE REVIEW

File: src/elm/Page/Device.elm [hunk 1]
```diff
@@ -667,14 +667,14 @@ handleKeyValueCommand session model cmd =
             DeviceAliasesUpdated newAliases
                 |> AstarteApi.updateDeviceAliases session.apiConfig model.deviceId (Dict.fromList [ ( key, value ) ])
 
-        ( Just device, Just (NewMetadata _ _), AskKeyValue.Confirm key value ) ->
+        ( Just device, Just (NewAttribute _ _), AskKeyValue.Confirm key value ) ->
             let
-                newMetadata =
-                    device.metadata
+                newAttribute =
```
Comment (davidebriani) [inline src/elm/Page/Device.elm:672]: `newAttribute` -> `newAttributes`

File: src/elm/Page/Device.elm [hunk 2]
```diff
@@ -698,14 +698,14 @@ handleSingleValueCommand session model cmd =
             DeviceAliasesUpdated newAliases
                 |> AstarteApi.updateDeviceAliases session.apiConfig model.deviceId (Dict.fromList [ ( key, value ) ])
 
-        ( Just device, Just (EditMetadataValue _ _ key), AskSingleValue.Confirm value ) ->
+        ( Just device, Just (EditAttributeValue _ _ key), AskSingleValue.Confirm value ) ->
             let
-                newMetadata =
-                    device.metadata
+                newAttribute =
```
Comment (davidebriani) [inline src/elm/Page/Device.elm:703]: `newAttribute` -> `newAttributes`

File: src/elm/Page/Device.elm [hunk 3]
```diff
@@ -837,7 +837,7 @@ view model flashMessages =
                 , Grid.row []
                     [ deviceInfoCard device Card.HalfWidth
                     , deviceAliasesCard device Card.HalfWidth
-                    , deviceMetadataCard device Card.HalfWidth
+                    , deviceAttributeCard device Card.HalfWidth
```
Comment (davidebriani) [inline src/elm/Page/Device.elm:840]: `deviceAttributeCard` -> `deviceAttributesCard`

File: src/elm/Page/Device.elm [hunk 4]
```diff
@@ -1634,10 +1634,10 @@ renderAliases aliases =
...
 
 
-renderMetadata : Dict String String -> Html Msg
-renderMetadata metadata =
-    if Dict.isEmpty metadata then
-        Card.simpleText "Device has no metadata"
+renderAttribute : Dict String String -> Html Msg
+renderAttribute attribute =
+    if Dict.isEmpty attribute then
+        Card.simpleText "Device has no attribute"
```
Comment (davidebriani) [inline src/elm/Page/Device.elm:1640]: `renderAttribute` -> `renderAttributes`
`attribute` -> `attributes`

File: src/elm/Types/Device.elm [hunk 5]
```diff
@@ -160,6 +160,6 @@ encodeCredentialsInhibited enabled =
     Encode.object [ ( "credentials_inhibited", Encode.bool enabled ) ]
 
 
-encodeMetadata : Dict String String -> Value
-encodeMetadata aliases =
-    Encode.object [ ( "metadata", Encode.dict identity Encode.string aliases ) ]
+encodeAttribute : Dict String String -> Value
+encodeAttribute attributes =
```
Comment (davidebriani) [inline src/elm/Types/Device.elm:164]: `encodeAttribute` -> `encodeAttributes`

File: src/react/DevicesPage.tsx [hunk 6]
```diff
@@ -66,30 +66,30 @@ const matchMetadata = (
   return true;
 };
 
-interface MatchedMetadataProps {
+interface MatchedAttributeProps {
   filters: DeviceFilters;
-  metadata: AstarteDevice['metadata'];
+  attribute: AstarteDevice['attributes'];
```
Comment (davidebriani) [inline src/react/DevicesPage.tsx:71]: `attribute` -> `attributes`

File: src/react/DevicesPage.tsx [hunk 7]
```diff
@@ -66,30 +66,30 @@ const matchMetadata = (
...
 }
 
-const MatchedMetadata = ({
+const MatchedAttribute = ({
```
Comment (davidebriani) [inline src/react/DevicesPage.tsx:74]: `MatchedAttribute` -> `MatchedAttributes`
`MatchedAttributeProps` -> `MatchedAttributesProps`

File: src/elm/AstarteApi.elm [hunk 8]
```diff
@@ -532,30 +532,30 @@ removeDeviceAlias apiConfig deviceId aliasTag resultMsg =
...
     Http.request
         { method = "PATCH"
         , headers = buildHeaders apiConfig.token
         , url = buildUrl apiConfig.secureConnection apiConfig.appengineUrl [ "v1", apiConfig.realm, "devices", deviceId ] []
-        , body = Http.stringBody "application/merge-patch+json" <| Encode.encode 0 <| Encode.object [ ( "data", Device.encodeMetadata metadata ) ]
+        , body = Http.stringBody "application/merge-patch+json" <| Encode.encode 0 <| Encode.object [ ( "data", Device.encodeAttribute attribute ) ]
         , expect = expectWhateverAstarteReply resultMsg
         , timeout = Nothing
         , tracker = Nothing
         }
```
Comment (davidebriani) [inline src/elm/AstarteApi.elm:545]: `updateDeviceAttribute ... attribute ...` -> `updateDeviceAttributes ... attributes ...`
`Device.encodeAttribute attribute` -> `Device.encodeAttributes attributes`

**Review by davidebriani — CHANGES_REQUESTED** [review]

**Review by davidebriani — APPROVED** [review]

File: CHANGELOG.md [hunk 9]
```diff
@@ -7,6 +7,8 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 ## [1.0.0-beta.2] - Unreleased
 ### Fixed
 - Hide login image column on small screen devices.
+### Changed
+- Parse device Attributes field in place of removed Metadata field.
```
Comment (bettio) [inline CHANGELOG.md:11]: I don't like it, this is uninteresting for users. "- Device metadata are called device attributes now. Astarte >= 1.0.0-beta.2 is required."

**Review by bettio — CHANGES_REQUESTED** [review]

**Review by bettio — APPROVED** [review]

**Pull request merged**